Added GCP Ocean Integration docs (#1159)
port-labs/port-docs: docs/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/installation.md

---

> sidebar_position: 1

Reviewer: Move this to be the first child under "Google cloud", like we do in similar integrations

---

# Installation

> The GCP exporter is deployed using Terraform on Google Cloud Cloud Run.

Reviewer: ```suggestion
The GCP integration is deployed using Terraform on Google Cloud Cloud Run.
```

> The GCP exporter is deployed using Terraform on Google Cloud Cloud Run.

Reviewer: google cloud cloud run? is this the correct number of clouds?

Author: GCP will be renamed in the future to Google Cloud (as Gut told me). Here it's really dumb so yeah i've changed to GCP

It uses our Terraform [Ocean](https://ocean.getport.io) Integration Factory [module](https://registry.terraform.io/modules/port-labs/integration-factory/ocean/latest) to deploy the exporter.

> ## Google Cloud infrastructure used by the Google Cloud exporter

Reviewer: ```suggestion
## Infrastructure
```

The Google Cloud exporter uses the following GCP infrastructure:

> - GCP Cloud Run;

Reviewer: Again, don't use `;` for list items, use `.`

- GCP PubSub Topic & Subscription;
- GCP Roles & Service accounts;
- GCP Cloud Assets:
  - GCP Cloud Asset Inventory
  - GCP Cloud Asset Feed  (Used for real-time data sync to Port)

## Prerequisites

- [Terraform](https://www.terraform.io/downloads.html) >= 0.15.0
- [A logged in gCloud CLI](https://cloud.google.com/sdk/gcloud)
- [Artifact Registry Image](https://cloud.google.com/artifact-registry/docs/docker/manage-images)
- [Permissions](#permissions)

## Artifact Registry Image

> In order to run the Cloud Run Service, it's mandatory to have a working Image. Currently our GHCR based images aren't supported by the Google Cloud's Cloud run platform, so a manuall install to Dockerhub\Artifact registry is required. In the guide we specify an Artifact registry approach, but similar DockerHub approach should result with the same results:

Reviewer: ```suggestion
In order to run the Cloud Run Service, it's mandatory to have a working Image. Currently our GHCR based images aren't supported by Google Cloud's Cloud run platform, so a manual installation to Dockerhub\Artifact registry is required. In the guide we specify an Artifact registry approach, but a similar DockerHub approach should yield the same results:
```

> 1. Create an Artifact Registry in GCP

Reviewer: 1. Change numbers to 1,2,3,4 and not just 1.
2. Add a `.` to the end of each item.
3. Since you're sharing commands that you want users to copy and execute, put them in a code block in a separate row so that theyre easy to copy. If you need help with this let me know

1. Pull our AMD based image from our GHCR registry: `docker pull ghcr.io/port-labs/port-ocean-gcp --platform amd64`
1. Tag this image like so: `docker tag ghcr.io/port-labs/port-ocean-gcp:latest <your_artifact_registry_/_dockerhub>/port-ocean-gcp:<your_version>`
1. Push the image to your artifact registry: `docker push <your_artifact_registry_/_dockerhub>/port-ocean-gcp:<your_version>`

## Permissions

In order to successfully deploy the GCP exporter, it's crucial to ensure that the user who deploys the integration in the GCP Organization has the appropriate access permissions.

> :::tip

Reviewer: The `tip` admonition is not closed so it goes on until the end of the page, close it with `:::`.

    cloudasset.feeds.delete
    cloudasset.feeds.get
    cloudasset.feeds.list
    cloudasset.feeds.update
    iam.roles.create
    iam.roles.delete
    iam.roles.get
    iam.roles.undelete
    iam.roles.update
    iam.serviceAccountKeys.get
    iam.serviceAccounts.actAs
    iam.serviceAccounts.create
    iam.serviceAccounts.delete
    iam.serviceAccounts.get
    pubsub.subscriptions.consume
    pubsub.subscriptions.create
    pubsub.subscriptions.delete
    pubsub.subscriptions.get
    pubsub.subscriptions.list
    pubsub.subscriptions.update
    pubsub.topics.attachSubscription
    pubsub.topics.create
    pubsub.topics.delete
    pubsub.topics.get
    pubsub.topics.list
    pubsub.topics.update
    resourcemanager.organizations.getIamPolicy
    resourcemanager.organizations.setIamPolicy
    run.operations.get
    run.services.create
    run.services.delete
    run.services.get
    serviceusage.services.use
    ```
    </details>
  - Go to the `Manage Resources` window at your organization's GCP
    - Click on the Organization
    - At the right side, Clock on `Add Principal`
    - Here, Enter your new service account principal, and grant it your newly created Role.

> ## Installation

Reviewer: This is quite a clunky installation method.. Today I discovered this link that takes you directly to the "Add data source" modal in your builder:
https://app.getport.io/settings/data-sources?section=EXPORTERS

Please use that and make the installation instructions shorter and simpler.

1. Login to [Port](https://app.getport.io) and browse to the [builder page](https://app.getport.io/settings)
2. Open the ingest modal by expanding one of the blueprints and clicking the ingest button on the blueprints.
3. Click on the GCP Exporter option under the Cloud Providers section:
4. Edit and copy the installation command.
5. Run the command in your terminal to deploy the GCP exporter.

> ## Further information

Reviewer: I would remove this section all together, or rename it to "Examples" with a link to the examples page.

    cloudasset.feeds.get
    cloudasset.feeds.list
    cloudasset.feeds.update
    iam.roles.create
    iam.roles.delete
    iam.roles.get
    iam.roles.undelete
    iam.roles.update
    iam.serviceAccountKeys.get
    iam.serviceAccounts.actAs
    iam.serviceAccounts.create
    iam.serviceAccounts.delete
    iam.serviceAccounts.get
    pubsub.subscriptions.consume
    pubsub.subscriptions.create
    pubsub.subscriptions.delete
    pubsub.subscriptions.get
    pubsub.subscriptions.list
    pubsub.subscriptions.update
    pubsub.topics.attachSubscription
    pubsub.topics.create
    pubsub.topics.delete
    pubsub.topics.get
    pubsub.topics.list
    pubsub.topics.update
    resourcemanager.organizations.getIamPolicy
    resourcemanager.organizations.setIamPolicy
    run.operations.get
    run.services.create
    run.services.delete
    run.services.get
    serviceusage.services.use
    ```
    </details>
  - Go to the `Manage Resources` window at your organization's GCP
    - Click on the Organization
    - At the right side, Clock on `Add Principal`
    - Here, Enter your new service account principal, and grant it your newly created Role.

## Installation walkthrough

> 1. Go to Port's [Data Sources](https://app.getport.io/settings/data-sources?section=EXPORTERS) and click on GCP

Reviewer: ```suggestion
1. Go to Port's [Data Sources](https://app.getport.io/settings/data-sources?section=integrationS) and click on GCP
```
```suggestion
1. Go to Port's [Data Sources](https://app.getport.io/settings/data-sources?section=EXPORTERS) and click on GCP.
```

2. Edit and copy the installation command.
3. Run the command in your terminal to deploy the GCP integration.

## Further Examples

> - Refer to the [examples](/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/) page for practical configurations and their corresponding blueprint definitions.

Reviewer: ```suggestion
Refer to the [examples](/build-your-software-catalog/sync-data-to-catalog/cloud-providers/gcp/examples/) page for practical configurations and their corresponding blueprint definitions.
```
